### PR TITLE
Full Python 3 Unicode identifier support using ANTLR 4.7-SNAPSHOT

### PR DESCRIPTION
**Pull request by bhamiltoncx**
The following is valid Python 3:

```
% cat test.py
𝛏𝟬 = '😀'
print(𝛏𝟬)

% python3 test.py
😀
```

Note that the identifier `𝛏𝟬` and the string literal `'😀'` both include Unicode values > `U+FFFF`.

However, the current `python3-parser` fails to parse Unicode values > `U+FFFF` due to https://github.com/antlr/antlr4/issues/276 :

```
% cat TestPython3Parser.java
import nl.bigo.pythonparser.Python3Parser;
import nl.bigo.pythonparser.Builder;

import java.nio.charset.StandardCharsets;
import java.nio.file.Paths;
import java.nio.file.Files;

public class TestPython3Parser {
  public static void main(String[] args) throws Exception {
    for (String arg : args) {
      byte[] argBytes = Files.readAllBytes(Paths.get(arg));
      String source = new String(argBytes, StandardCharsets.UTF_8);
      System.out.println(source);
      System.out.println(new Builder.Tree(source).toStringASCII());
    }
  }
}

% cat test.py
𝛏𝟬 = '😀'
print(𝛏𝟬)

% javac -cp /home/beng/.m2/repository/nl/big-o/python3-parser/0.1.0/python3-parser-0.1.0.jar TestPython3Parser.java

% java -cp /home/beng/.m2/repository/org/antlr/antlr4-runtime/4.3/antlr4-runtime-4.3.jar:/home/beng/.m2/repository/nl/big-o/python3-parser/0.1.0/python3-parser-0.1.0.jar:. TestPython3Parser test.py
𝛏𝟬 = '😀'
print(𝛏𝟬)

Exception in thread "main" java.lang.RuntimeException: source: null, line: 1, index: 0, error message: mismatched input '?' expecting {<EOF>, 'def', 'return', 'raise', 'from', 'import', 'global', 'nonlocal', 'assert', 'if', 'while', 'for', 'try', 'with', 'lambda', 'not', 'None', 'True', 'False', 'class', 'yield', 'del', 'pass', 'continue', 'break', NEWLINE, NAME, STRING_LITERAL, BYTES_LITERAL, DECIMAL_INTEGER, OCT_INTEGER, HEX_INTEGER, BIN_INTEGER, FLOAT_NUMBER, IMAG_NUMBER, '...', '*', '(', '[', '+', '-', '~', '{', '@'} null
        at nl.bigo.pythonparser.DescriptiveBailErrorListener.syntaxError(DescriptiveBailErrorListener.java:22)
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:67)
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:561)
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:150)
        at nl.bigo.pythonparser.Python3Parser.file_input(Python3Parser.java:326)
        at nl.bigo.pythonparser.Builder$Tree.toStringASCII(Builder.java:90)
        at TestPython3Parser.main(TestPython3Parser.java:14)
```

I fixed https://github.com/antlr/antlr4/issues/276 and ANTLR 4.7 now supports Unicode literals > `U+FFFF` (as long as you use the new `CharStreams` API to read the data):

https://github.com/antlr/antlr4/blob/master/doc/unicode.md

This PR updates `python3-parser` to support Unicode values > `U+FFFF` in identifiers and strings:

```
% javac -cp /home/beng/.m2/repository/nl/big-o/python3-parser/0.1.0/python3-parser-0.1.0.jar TestPython3Parser.java

% java -cp /home/beng/.m2/repository/org/antlr/antlr4-runtime/4.7-SNAPSHOT/antlr4-runtime-4.7-SNAPSHOT.jar:/home/beng/.m2/repository/nl/big-o/python3-parser/0.1.0/python3-parser-0.1.0.jar:. TestPython3Parser test.py                  
𝛏𝟬 = '😀'
print(𝛏𝟬)

'- file_input
   |- stmt
   |  '- simple_stmt
   |     |- small_stmt
   |     |  '- expr_stmt
   |     |     |- testlist_star_expr
   |     |     |  '- test
   |     |     |     '- or_test
   |     |     |        '- and_test
   |     |     |           '- not_test
   |     |     |              '- comparison
   |     |     |                 '- star_expr
   |     |     |                    '- expr
   |     |     |                       '- xor_expr
   |     |     |                          '- and_expr
   |     |     |                             '- shift_expr
   |     |     |                                '- arith_expr
   |     |     |                                   '- term
   |     |     |                                      '- factor
   |     |     |                                         '- power
   |     |     |                                            '- atom
   |     |     |                                               '- 𝛏𝟬
   |     |     |- =
   |     |     '- testlist_star_expr
   |     |        '- test
   |     |           '- or_test
   |     |              '- and_test
   |     |                 '- not_test
   |     |                    '- comparison
   |     |                       '- star_expr
   |     |                          '- expr
   |     |                             '- xor_expr
   |     |                                '- and_expr
   |     |                                   '- shift_expr
   |     |                                      '- arith_expr
   |     |                                         '- term
   |     |                                            '- factor
   |     |                                               '- power
   |     |                                                  '- atom
   |     |                                                     '- string
   |     |                                                        '- '😀'
   |     '- <NEWLINE>
   |- stmt
   |  '- simple_stmt
   |     |- small_stmt
   |     |  '- expr_stmt
   |     |     '- testlist_star_expr
   |     |        '- test
   |     |           '- or_test
   |     |              '- and_test
   |     |                 '- not_test
   |     |                    '- comparison
   |     |                       '- star_expr
   |     |                          '- expr
   |     |                             '- xor_expr
   |     |                                '- and_expr
   |     |                                   '- shift_expr
   |     |                                      '- arith_expr
   |     |                                         '- term
   |     |                                            '- factor
   |     |                                               '- power
   |     |                                                  |- atom
   |     |                                                  |  '- print
   |     |                                                  '- trailer
   |     |                                                     |- (
   |     |                                                     |- arglist
   |     |                                                     |  '- argument
   |     |                                                     |     '- test
   |     |                                                     |        '- or_test
   |     |                                                     |           '- and_test
   |     |                                                     |              '- not_test
   |     |                                                     |                 '- comparison
   |     |                                                     |                    '- star_expr
   |     |                                                     |                       '- expr
   |     |                                                     |                          '- xor_expr
   |     |                                                     |                             '- and_expr
   |     |                                                     |                                '- shift_expr
   |     |                                                     |                                   '- arith_expr
   |     |                                                     |                                      '- term
   |     |                                                     |                                         '- factor
   |     |                                                     |                                            '- power
   |     |                                                     |                                               '- atom
   |     |                                                     |                                                  '- 𝛏𝟬
   |     |                                                     '- )
   |     '- <NEWLINE>
   '- <EOF>
```

Note that this adds a dependency on ANTLR4 4.7-SNAPSHOT, which is not in Maven, so you will need to locally clone antlr4 and install it via `mvn -DskipTests=true install`.

/cc @parrt

### CODE REVIEW

File: src/main/antlr4/nl/bigo/pythonparser/Python3.g4 [hunk 1]
```diff
@@ -867,7 +867,7 @@ RIGHT_SHIFT_ASSIGN : '>>=';
 POWER_ASSIGN : '**=';
 IDIV_ASSIGN : '//=';
 
-SKIP
+SKIPPABLE
```
Comment (bhamiltoncx) [inline src/main/antlr4/nl/bigo/pythonparser/Python3.g4]: FYI, `SKIP` is now a reserved word in ANTLR 4.7.

**Comment by bkiers** [comment]
Thanks @bhamiltoncx, that sure cleans it up!

I'll merge once 4.7 is in Maven Central and TravisCI is happy with the build.

**Comment by bhamiltoncx** [comment]
Sounds good!

**Comment by parrt** [comment]
hooray!

**Comment by bhamiltoncx** [comment]
OK, 4.7 is released and ought to make its way to Maven Central before too long. Updated PR.

**Comment by bkiers** [comment]
Nice @bhamiltoncx!

**Comment by bhamiltoncx** [comment]
Cheers!

**Pull request merged**